packages/adminrouter/extra/src/test-harness/tests/test_generic.py

```diff
@@ -0,0 +1,478 @@
+# Copyright (C) Mesosphere, Inc. See LICENSE file for details.
```

Comment:
I am using `@pytest.mark.parametrize()` using the data from YAML file. So you get something like this:
```
test-harness/tests/test_generic.py::TestAgentGeneric::test_if_upstream_request_is_correct[/system/health/v1/-/system/health/v1/-HTTP/1.0] PASSED
test-harness/tests/test_generic.py::TestAgentGeneric::test_if_upstream_request_is_correct[/system/health/v1-/system/health/v1-HTTP/1.0] PASSED
test-harness/tests/test_generic.py::TestAgentGeneric::test_if_upstream_request_is_correct[/system/v1/logs/v1/foo/bar-/foo/bar-HTTP/1.1] PASSED
test-harness/tests/test_generic.py::TestAgentGeneric::test_if_upstream_request_is_correct[/system/v1/logs/v1/-/-HTTP/1.1] PASSED
```
Each entry in YAML file is basically a separate tests.